From 8ab4b5063e3c4480006f0d14cc03a73bbf3f2878 Mon Sep 17 00:00:00 2001
Message-Id: <8ab4b5063e3c4480006f0d14cc03a73bbf3f2878.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 30 Dec 2014 20:17:37 +0530
Subject: [PATCH 83/95] MUST_REBASE [VPG]: video/adf/intel: Respect ADF kernel
 paramter.

Verify "enable_intel_adf" to help adf driver to decide if ADF drivers
should be initilized or not for a given platform

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need
to work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5501
Change-Id: I389d1c560e9a26dede1f62feb54e19874cca8004
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/video/adf/intel/core/intel_adf_drv.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/adf/intel/core/intel_adf_drv.c b/drivers/video/adf/intel/core/intel_adf_drv.c
index 0bd77ae..67ce4d8 100644
--- a/drivers/video/adf/intel/core/intel_adf_drv.c
+++ b/drivers/video/adf/intel/core/intel_adf_drv.c
@@ -30,9 +30,17 @@ struct intel_adf_context *adf_ctx;
 static int intel_adf_init(void)
 {
 	struct pci_dev *i915_pci_dev;
+	int adf_enabled;
 	pr_err("ADF: %s\n", __func__);
 
 	i915_pci_dev = i915_adf_get_pci_dev();
+
+	adf_enabled = i915_adf_driver_initialize();
+	if (!adf_enabled) {
+		pr_err("%s:adf disabled in kernel parameter\n", __func__);
+		return -EBUSY;
+	}
+
 	adf_ctx = intel_adf_context_create(i915_pci_dev);
 	if (IS_ERR(adf_ctx)) {
 		pr_err("%s:failed to create ADF context\n", __func__);
-- 
1.7.9.5

